rust/agents/relayer/src/msg/processor.rs

```diff
@@ -106,7 +296,6 @@ impl ProcessorExt for MessageProcessor {
                 app_context,
             );
             self.send_channels[&destination].send(Box::new(pending_msg) as QueueOperation)?;
-            self.message_nonce += 1;
         } else {
             tokio::time::sleep(Duration::from_secs(1)).await;
         }
```

Comment:
iiuc this whole loop is similar to what I describe here https://discord.com/channels/935678348330434570/1238095499312234597/1241053215529898095 - this task won't ever block and yield to the executor until all known messages in the db have been processed, and we finally hit this sleep when the high and low iterators indicate the next message isn't indexed yet.

Wdyt of some basic logic in this loop where we https://docs.rs/tokio/latest/tokio/task/fn.yield_now.html every once in a while? Otherwise I'm worried that the submitter may not be able to get scheduled if submitters are hogging execution

Same for the backend cursor tbh https://discord.com/channels/935678348330434570/1238095499312234597/1241052971375132724, I feel like yielding may lower start time considerably